types/src/messages/client_as_out.rs

```diff
@@ -201,37 +196,47 @@ impl Verifiable for VerifiableConnectionPackage {
 }
 
 #[derive(Debug, TlsSerialize, TlsDeserializeBytes, TlsSize)]
-pub struct FinishUserRegistrationParamsTbsIn {
-    pub client_id: AsClientId,
+pub struct InitUserRegistrationParamsIn {
+    pub client_payload: ClientCredentialPayload,
     pub queue_encryption_key: RatchetEncryptionKey,
     pub initial_ratchet_secret: RatchetSecret,
-    pub connection_packages: Vec<ConnectionPackageIn>,
-    pub opaque_registration_record: OpaqueRegistrationRecord,
     pub encrypted_user_profile: EncryptedUserProfile,
 }
 
+impl NoAuth for InitUserRegistrationParamsIn {
+    fn into_verified(self) -> VerifiedAsRequestParams {
+        VerifiedAsRequestParams::InitUserRegistration(self)
+    }
+}
+
 #[derive(Debug, TlsDeserializeBytes, TlsSize)]
-pub struct FinishUserRegistrationParamsIn {
-    payload: FinishUserRegistrationParamsTbsIn,
+pub struct AsPublishConnectionPackagesParamsIn {
+    payload: AsPublishConnectionPackagesParamsTbsIn,
     signature: Signature,
 }
 
-impl ClientCredentialAuthenticator for FinishUserRegistrationParamsIn {
-    type Tbs = FinishUserRegistrationParamsTbsIn;
+impl ClientCredentialAuthenticator for AsPublishConnectionPackagesParamsIn {
+    type Tbs = AsClientId;
 
     fn client_id(&self) -> AsClientId {
         self.payload.client_id.clone()
     }
 
     fn into_payload(self) -> VerifiedAsRequestParams {
-        VerifiedAsRequestParams::FinishUserRegistration(self.payload)
+        VerifiedAsRequestParams::PublishConnectionPackages(self.payload)
     }
 
     fn signature(&self) -> &Signature {
         &self.signature
     }
 
-    const LABEL: &'static str = "Finish User Registration Parameters";
+    const LABEL: &'static str = "Publish ConnectionPackages Parameters";
```

Comment:
It might make sense to make this point to the same string as `Signable` implementation of `AsPublishConnectionPackagesParamsTbs`.

Reply:
Good point. Fixed.